### PR TITLE
[Base64_rfc2045.decode] should always progresses

### PR DESCRIPTION
**Pull request by tiash**
Currently [Base64_rfc2045.decode] will get stuck on many unexpected characters, e.g. a single "\r".
This pull request fixes this to **always** make forward progress and to only discard 1 character, this allows graceful recovery for most malformed inputs (similar to ``Core.Base64``).

This pull request should make #33 unecessary, as white-space can be ignored by the client by ignoring `` `Malformed str `` where `` str `` consists of white-space only.

It's should be possible to make the parser stricter now and to enforce the encode/decode isomorphism while supporting more permissive use cases (e.g. parsing email bodies robustly)

Finally, the changes the the CRLF handling code where needed to make positive progress around lone "\r" characters (the original version was hard to reason about and fix to not also skip the character after "\r")

### CODE REVIEW

File: src/base64_rfc2045.ml [hunk 1]
```diff
@@ -186,9 +141,9 @@ and t_flush {quantum; size; buffer} =
       unsafe_set_chr buffer 0 (unsafe_chr ((quantum lsr 8) land 255)) ;
       unsafe_set_chr buffer 1 (unsafe_chr (quantum land 255)) ;
       `Flush {quantum; size; buffer= Bytes.sub buffer 0 2}
-  | _ -> malformed buffer 0 0 3
+  | _ -> assert false
```
Comment (dinosaure) [inline src/base64_rfc2045.ml]: At least, a comment to explain why this case is not occur is needed here.

Comment (tiash) [reply src/base64_rfc2045.ml]: My bad, fixed now.
This can never happen as ``size`` will only ever be in the range``[0..3]``.

**Comment by dinosaure** [comment]
I apply your request on `Wrong_padding` case too and add some tests to ensure:
* Current decoder can decode malformed input
* Notify user about malformed input (`Malformed _` or `Wrong_padding`)
* As user wants (strict or less paranoid), have expected behavior:

* `test` provides an implementation which raises an exception when we retrieve `Malformed _` or `Wrong_padding`
* `test` provides a relaxed implementation where we just move on a error cases

**Comment by tiash** [comment]
I like it! Thanks!

**Comment by dinosaure** [comment]
Ah and I ran this version with [`mrmime`](https://github.com/mirage/mrmime.git) over all my e-mails and did not catch any weird bugs. Waiting the CI and ready to merge!

**Pull request merged**